### ISSUE TITLE
Panic when attempting to decode an empty image as a "pure barcode"

### ISSUE THREAD

**Issue by dlenski** [rxing-core/rxing#58]
Attempting to decode an empty image (e.g. https://github.com/dlenski/python-zxing/blob/master/test/barcodes/empty.png) as a pure barcode results in a panic at https://github.com/rxing-core/rxing/blob/v0.6.2/src/oned/one_d_reader.rs#L190

Example with `rxing-cli`:

```
$ rxing-cli test/barcodes/empty.png decode --pure-barcode true
thread 'main' panicked at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rxing-0.6.2/src/oned/one_d_reader.rs:190:17:
index out of bounds: the len is 0 but the index is 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

In non-pure-barcode mode, it gives a more usual `NotFoundException`:

```
$ rxing-cli test/barcodes/empty.png decode --pure-barcode false
Error while attempting to locate barcode in 'test/barcodes/empty.png': NotFoundException
```

**Comment by hschimke** [comment]
I’ll look into this. Thank you.

**Comment by hschimke** [comment]
This should now be fixed. Please let me know/reopen the issue if you see anything else. Thanks!

Fixed in: v0.6.3
Published: December 18, 2024
